ray-operator/controllers/ray/utils/dashboard_httpclient.go

```diff
@@ -70,70 +63,38 @@ type RayDashboardClient struct {
 	BaseDashboardClient
 }
 
-func FetchDashboardAgentURL(ctx context.Context, log *logr.Logger, cli client.Client, rayCluster *rayv1alpha1.RayCluster) (string, error) {
-	dashboardAgentService := &corev1.Service{}
-	dashboardAgentServiceName := CheckName(GenerateDashboardServiceName(rayCluster.Name))
-	if err := cli.Get(ctx, client.ObjectKey{Name: dashboardAgentServiceName, Namespace: rayCluster.Namespace}, dashboardAgentService); err != nil {
-		return "", err
-	}
-
-	log.V(1).Info("fetchDashboardAgentURL ", "dashboard agent service found", dashboardAgentService.Name)
-	// TODO: compare diff and reconcile the object. For example. ServiceType might be changed or port might be modified
-	servicePorts := dashboardAgentService.Spec.Ports
-
-	dashboardPort := int32(-1)
-
-	for _, servicePort := range servicePorts {
-		if servicePort.Name == DefaultDashboardAgentListenPortName {
-			dashboardPort = servicePort.Port
-			break
-		}
-	}
-
-	if dashboardPort == int32(-1) {
-		return "", fmtErrors.Errorf("dashboard port not found")
-	}
-
-	domainName := GetClusterDomainName()
-	dashboardAgentURL := fmt.Sprintf("%s.%s.svc.%s:%v",
-		dashboardAgentService.Name,
-		dashboardAgentService.Namespace,
-		domainName,
-		dashboardPort)
-	log.V(1).Info("fetchDashboardAgentURL ", "dashboardURL", dashboardAgentURL)
-	return dashboardAgentURL, nil
-}
-
-func FetchDashboardURL(ctx context.Context, log *logr.Logger, cli client.Client, rayCluster *rayv1alpha1.RayCluster) (string, error) {
+// FetchHeadServiceURL fetches the URL that consists of the FQDN for the RayCluster's head service
+// and the port with the given port name (defaultPortName).
+func FetchHeadServiceURL(ctx context.Context, log *logr.Logger, cli client.Client, rayCluster *rayv1alpha1.RayCluster, defaultPortName string) (string, error) {
 	headSvc := &corev1.Service{}
 	headSvcName := GenerateServiceName(rayCluster.Name)
 	if err := cli.Get(ctx, client.ObjectKey{Name: headSvcName, Namespace: rayCluster.Namespace}, headSvc); err != nil {
 		return "", err
 	}
 
-	log.V(3).Info("fetchDashboardURL ", "dashboard service found", headSvc.Name)
```

Comment:
What's V(3) and why remove it?

Reply:
This is used to specify verbosity. Higher verbosity means that this information is less important. `log.Info` is equivalent to `log.V(0).Info`. Hence, I just increased the importance of the message."

Reply:
Thanks, makes sense!